### ISSUE TITLE
DisallowYodaConditions sniff - constant support?

### ISSUE THREAD

**Issue by michalbundyra** [squizlabs/PHP_CodeSniffer#2830]
Example:

```php
if (PHP_EOL === $a) {}
if (DateTime::ATOM === $a) {}
```

Both, imo, should be considered as yoda conditions and sniff `DisallowYodaConditionsSniff` should catch them.

**Comment by rodrigoprimo** [comment]
Two more false negatives that I found while working on a PR to improve code coverage for this sniff:

```php
[1, 2] === $value;
[null] === $value;
```

**Comment by jrfnl** [comment]
@rodrigoprimo I agree that those would be false negatives, but they are a different issue to solve (recognizing constants vs recognizing short arrays), so should probably be reported in their own ticket. (in the new repo)

**Comment by rodrigoprimo** [comment]
@jrfnl, do you think it makes sense to consolidate both issues (recognizing constants and recognizing some sort arrays) into a single false negatives issue in the new repo? I ask this because I believe it is likely we will find more false negatives for this sniff. If you prefer a separate issue, I will go ahead a create a new issue in the new repo just for the false negatives that I reported above.

**Comment by jrfnl** [comment]
> @jrfnl, do you think it makes sense to consolidate both issues (recognizing constants and recognizing some sort arrays) into a single false negatives issue in the new repo?

As I said above, no, these are separate issues. While they _could_ possibly be solved in one PR, it would also be perfectly valid to address each in a separate PR. Having lots of loosely related, but distinctly different issues in one ticket makes for tickets which never get addressed or which only get partially addressed, meaning the ticket can never be closed.
